src/fileActions.jsx

```diff
@@ -625,3 +625,31 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        // TODO: is this a good value?
+        max_read_size: 150 * 1024 * 1024,
+        external: {
+            "content-disposition": `attachment; filename=\\"${selected.name}\\"`,
+            "content-type": "application/octet-stream"
+        }
+    }));
+
+    const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
+    const url = prefix + "?" + query;
+    return new Promise((resolve, reject) => {
+        const iframe = document.createElement("iframe");
+        iframe.setAttribute("src", url);
```

Comment:
Downloading a file through an iframe is something I haven't seen before. How does this handle chunking? There doesn't seem to be any kind of progress information in the code -- is that handled by the browser? Does the fsread1 channel give a proper `Content-Length:` so that the browser knows how much data to expect?

There should be a test case with a huge file (sparse with `truncate` should work fine), to make sure that this doesn't overflow RAM and does chunking correctly.

Reply:
So we do the same in SoS report to download reports. I'll have to see if we get `Content-length`. I can easily add a test for a huge fle.

Reply:
This is now resolved in Cockpit-ws via https://github.com/cockpit-project/cockpit/pull/19889